urllib3/connectionpool.py

```diff
@@ -568,25 +590,19 @@ def urlopen(self, method, url, body=None, headers=None, retries=None,
             # Close the connection. If a connection is reused on which there
             # was a Certificate error, the next request will certainly raise
             # another Certificate error.
-            if conn:
-                conn.close()
-                conn = None
+            conn = self._cleanup_conn(conn, not release_conn)
```

Comment:
... Then call it with

```python
conn = self._cleanup_conn(conn, replace_conn=not release_conn)
```

It still has the negation, but at least it's not negating into the same variable name and you get context in the call.